Add MongoDB guide, MySQL Cloud SQL guide and other 7.0 docs updates

r0mant opened the pull request:
New content for 7.0:

* MongoDB guide.
* MySQL Cloud SQL guide.
* HA guide.
* Update guides to new `tsh db connect` syntax.
* Various other cleanups.

r0mant committed:
MySQL GCP guide, MongoDB guide, and other 7.0 docs updates

inertial-frame commented on docs/pages/database-access/guides/ha.mdx:
```diff
@@ -0,0 +1,97 @@
+
+You can deploy Database Access in a highly available configuration in a couple
+of common ways: separate replicas and combined replicas. Both of those revolve
+around pointing multiple Database Services to the same database instance.
```
One general request I have (that is in no way related to the other great changes here) involves using some of the new includes: `docs/pages/includes` (and factoring out reused configuration examples, warnings, notes, and steps into new ones as well).

We've been trying to add backup warnings here, for example. It's an obvious best practice but a good idea to include (almost all docs do this). We will also eventually upgrade all of our includes using some new UI/UX components:

1. https://github.com/gravitational/next/issues/275
2. https://github.com/gravitational/next/issues/298

This will not only make our message more consistent, it makes reviewing changes a little less stressful since it reduces the amount of text, also makes it easier to uniformly update the same items.

r0mant replied:
Good call, I was thinking about making some partials as well. I'll take a look if anything can be factored out. The issue is that the commands differ a little bit between the guides - do you know if these includes can be parameterized?

r0mant replied:
I've moved some duplicated sections into includes. Others can't be factored out easily without some parameterization support.

webvictim commented on docs/pages/database-access/guides/ha.mdx:
```diff
@@ -0,0 +1,97 @@
+outage), Teleport will automatically retry to connect via other agents.
+
+<Admonition type="note" title="">
+  Teleport does not automatically reconnect if the agent goes down in the
+  middle of a database session.
```
Does this mean that the Teleport agent doesn't automatically reconnect to the database, that the database client won't automatically reconnect to Teleport, or both?

r0mant replied:
I meant Teleport but I agree it might be a bit confusing since client may attempt to reconnect itself, so I just removed this note.

webvictim commented on docs/pages/database-access/guides/mongodb-self-hosted.mdx:
````diff
@@ -0,0 +1,400 @@
+  $ tctl auth sign --format=mongodb --host=mongo2.example.com --out=mongo2 --ttl=8760h
+  ```
+
+  Each command will create 2 files: `mongo{1,2}.cas` with Teleport's certificate
````
```suggestion
  Each command will create 2 files: `mongo1.cas` and `mongo2.cas` with Teleport's certificate
```

Might be better to be explicit about the filenames here - some people may not be familiar with the `bash` `{1,2}` syntax format.

webvictim commented on docs/pages/database-access/guides/mongodb-self-hosted.mdx:
```diff
@@ -0,0 +1,400 @@
+  enabled: "yes"
+  # Set public address proxy will be reachable at.
+  #
+  # MongoDB clients are connecting to regular web proxy endpoint.
```
```suggestion
  # MongoDB clients connect to Teleport's regular web proxy endpoint.
```

webvictim commented on docs/pages/database-access/guides/mongodb-self-hosted.mdx:
````diff
@@ -0,0 +1,400 @@
+EOF
+```
+
+Create the user assigned the `db` role we've just created:
````
```suggestion
Create a user assigned the `db` role we've just created:
```

webvictim commented on docs/pages/database-access/guides/mongodb-self-hosted.mdx:
```diff
@@ -0,0 +1,400 @@
+
+### Create Role and User
+
+Create the role that will allow a user to connect to any database using any
```
```suggestion
Create a role that will allow a user to connect to any database using any
```

webvictim commented on docs/pages/database-access/reference/cli.mdx:
```diff
@@ -101,29 +101,49 @@ $ tsh db logout example
 | Flag | Description |
 | - | - |
-| `--db` | Database server name to print configuration for. Required, if logged into multiple databases simultaneously. |
+| `--format` | Output format: text is default, `cmd` to print native database client connect command. |
```
```suggestion
| `--format` | Output format: `text` is default, `cmd` to print native database client connect command. |
```

webvictim commented on docs/pages/database-access/guides/mysql-cloudsql.mdx:
````diff
@@ -0,0 +1,292 @@
+Download the appropriate version of Teleport for your platform from our
+[downloads page](https://goteleport.com/teleport/download).
+
+Teleport requires a valid TLS certificate to operate and can fetch one automatically
+using Let's Encrypt [ACME](https://letsencrypt.org/how-it-works/) protocol.
+
+We will assume that you have configured DNS records for `teleport.example.com`
+and `*.teleport.example.com` to point to the Teleport node.
+
+Let's generate a Teleport config with ACME enabled:
+
+```bash
+$ teleport configure --cluster-name=teleport.example.com --acme --acme-email=alice@example.com -o file
+```
+
+<Admonition
+  type="note"
+  title="Web Proxy Port"
+>
+  Teleport uses [TLS-ALPN-01](https://letsencrypt.org/docs/challenge-types/#tls-alpn-01)
+  ACME challenge to validate certificate requests which only works on port `443`.
+  As such, in order to use ACME for certificate management, web proxy needs to
+  be accessible on port `443`.
+</Admonition>
````
As per @inertial-frame it'd be good if this whole ACME section was something included rather than duplicated content, as I know we have these same instructions elsewhere in the docs and maintenance will be cumbersome when they all need to be updated individually.

webvictim reviewed (approved):
LGTM, just a few nits.

r0mant commented on docs/pages/database-access/reference/cli.mdx:
````diff
@@ -40,17 +40,17 @@ $ teleport db start \
 
 ```shell
-tctl auth sign --format=tls --user=access-plugin-mattermost --out=auth --ttl=2190h
+$ tctl auth sign --format=db --host=db.example.com --out=db --ttl=8760h
````
@inertial-frame FYI this command was incorrectly changed in https://github.com/gravitational/teleport/pull/7158, I reverted it back so it makes sense for database access.

inertial-frame replied:
After some initial discussion, we agreed on using 3 months for our examples requiring long-lived certs. Confirming that we want all DB examples to be `8760h`? 

May be a few places that we'd want to revert that.

r0mant replied:
Hmm, I'm okay changing it though I'm wondering if 3 months may be a bit too aggressive for databases. Effectively means they would have to rotate secrets for all their dbs every 3 months which may be annoying (esp. since updating database certs requires restarting the database).

As an option we could change to 3 months as a recommendation (they can always choose their own TTL after all) and add a note with what I wrote above.

inertial-frame replied:
@r0mant - Yeah that makes sense. Anecdotally, rotating **6 months** or **annually** seems fairly common.

I don't believe GDPR or other compliance standards typically define a rotation lifecycle or timeframe. A quick bit of research shows **6 months** or **1 year** as fairly common.

We're adding an article that will be dedicated to short-lived TTL (excluding DB's and other mainstays) which will include basic commands for inspecting, revoking, renewing, etc. (we can then point to that article/section throughout the docs).

I'm wondering if for these long-lived certs if it's better to use a timeframe that aligns with the above (probably anywhere between **3 months to a year**, with a slight preference for shorter durations) but note that these are examples. Maybe this could be moved into an include as an Admonition?

r0mant replied:
@inertial-frame Okay, I changed it back to 2160h and added a note about picking TTL.

inertial-frame commented on docs/pages/database-access/guides/mysql-self-hosted.mdx:
```diff
@@ -7,54 +7,18 @@ description: How to configure Teleport Database Access with self-hosted MySQL
-# for host db.example.com with a 3-month validity period.
-tctl auth sign --format=db --host=db.example.com --out=server --ttl=2190h
+# for host db.example.com with a 1-year validity period.
+$ tctl auth sign --format=db --host=db.example.com --out=server --ttl=8192h
```
Double-checking: we're using `8192h` here but `8760h` elsewhere in this PR. Is there one value we should use? Are these values specific to one of the DB's but not the others?

r0mant replied:
Whoops, good catch - that was meant to be a year so 8760h of course :) I'll update once we settle on the value as discussed in the other thread.

inertial-frame commented on docs/pages/database-access/guides/mongodb-self-hosted.mdx:
````diff
@@ -0,0 +1,344 @@
+  hostname, run:
+
+  ```bash
+  $ tctl auth sign --format=mongodb --host=mongo.example.com --out=mongo --ttl=8760h
````
Double-checking: we're using `8760h` here but `8192h` elsewhere in this PR. Is there one value we should use? Are these values specific to one of the DB's but not the others?

inertial-frame commented on docs/pages/database-access/guides/ha.mdx:
```diff
@@ -0,0 +1,92 @@
+of common ways: separate replicas and combined replicas. Both of those revolve
+around pointing multiple Database Services to the same database instance.
+
+## Separate Replicas
```
Same casing per previous comments.

inertial-frame commented on docs/pages/database-access/guides/ha.mdx:
````diff
@@ -0,0 +1,92 @@
+With this configuration, both services will appear as two separate entries in
+`tsh db ls` output and you will have to pick one explicitly when connecting:
+
+```bash
````
I think we're still in the process of creating dedicated code snippet sections:

1. Should automatically add the `$`.
2. Should prevent `$` from being copied (annoying to copy this on ZSH for example).
3. Will hopefully more cleanly divide "output" examples and "input" ones - at least I've requested this.

We'll likely want to revamp all of the code snippets once this done in: https://github.com/gravitational/next/issues/275

r0mant replied:
Let me know when this change is through and I can update the snippets here.

inertial-frame commented on docs/pages/database-access/guides/mongodb-self-hosted.mdx:
```diff
@@ -0,0 +1,348 @@
+When configuring a replica set, make sure to do it for each member and use
+secrets generated for the particular server.
+
+<Admonition type="note" title="">
```
Stylistic comment: I think `title` is optional, you can either leave it out or add whatever you want. We often will just add the `type`. 

Thanks for using these!

inertial-frame commented on docs/pages/database-access/guides/mongodb-self-hosted.mdx:
```diff
@@ -0,0 +1,348 @@
+
+Teleport Database Access for MongoDB is available starting from `7.0` release.
+
+(!docs/pages/includes/database-access/start-auth-proxy.mdx!)
```
Thanks for refactoring some of these into includes!

inertial-frame commented on docs/pages/database-access/guides/mongodb-self-hosted.mdx:
````diff
@@ -0,0 +1,348 @@
+You can optionally specify the database name and the user to use by default
+when connecting to the database instance:
+
+```sh
````
I think `bash` and `sh` render the same but not `bsh`. As you know, the values we populate in the code blocks determines the coloring when they render. I've gone through and tried to use `bash` universally just to be sure.

It'll also help us to quickly GREP on all code snippets in the future.

inertial-frame commented on docs/pages/database-access/guides/mysql-cloudsql.mdx:
```diff
@@ -0,0 +1,254 @@
+
+(!docs/pages/includes/database-access/start-auth-proxy.mdx!)
+
+### Create User
```
Same heading comment applies here to `###` blocks as well. Thanks!

inertial-frame commented on docs/pages/database-access/guides/mysql-self-hosted.mdx:
```diff
@@ -7,54 +7,20 @@ description: How to configure Teleport Database Access with self-hosted MySQL
-# for host db.example.com with a 3-month validity period.
-tctl auth sign --format=db --host=db.example.com --out=server --ttl=2190h
+# for host db.example.com with a 1-year validity period.
+$ tctl auth sign --format=db --host=db.example.com --out=server --ttl=2190h
```
`2190` is 3-months.

Per our conversation, I'd think 3-moths, 6-months, or 12-months would be appropriate with a preference for 3-months or 6-months.

inertial-frame commented on docs/pages/database-access/guides/mysql-self-hosted.mdx:
````diff
@@ -7,54 +7,20 @@ description: How to configure Teleport Database Access with self-hosted MySQL
 ```
 
-Flag descriptions:
+(!docs/pages/includes/database-access/ttl-note.mdx!)
````
Great idea.

We can also add includes into code blocks too!

inertial-frame commented on docs/pages/database-access/guides/postgres-aws.mdx:
```diff
@@ -76,52 +76,12 @@ GRANT rds_iam TO alice;
+## Setup Teleport Auth and Proxy Services
 
-Teleport Database Access is available starting from `6.0` release.
+Teleport Database Access for PostgreSQL on AWS is available starting from `6.0`
```
Might rephrase this to: `Teleport Database Access for PostgreSQL on AWS is available 6.0+...` 

Previously we had one or two comments about versioning that seemed to indicate that the functionality was available in only one version (and not subsequent ones) or that needed to better indicate this for other reasons.

r0mant replied:
I think that "starting from" indicates clearly enough that it's available in all subsequent releases as well but let me know if you think otherwise.

inertial-frame commented on docs/pages/database-access/guides/postgres-self-hosted.mdx:
```diff
@@ -274,18 +200,21 @@ when connecting to the database instance:
-When logging into a PostgreSQL database, `tsh` automatically configures a section
-in the [connection service file](https://www.postgresql.org/docs/current/libpq-pgservice.html)
-with the name of `<cluster-name>-<database-service-name>`.
+Once logged in, connect to the database:
```
Overall, the steps are great. I love the brevity. You use one or two sentences, then a command.

r0mant replied:
Yeah I like how the new `tsh db connect` flow turned out, simpler than it was before and more uniform.

inertial-frame commented on docs/pages/database-access/introduction.mdx:
```diff
@@ -43,18 +43,56 @@ with Github, execute a few SQL queries and observe them in the audit log:
+## General Guides
+
+<TileSet>
+  <Tile icon="window" title="GUI clients" href="./guides/gui-clients.mdx">
```
Verifying that the icons should be `window` not `database`.

r0mant replied:
Yep, like mentioned above window icon looks like a GUI application so seems fitting.

inertial-frame commented on docs/pages/includes/database-access/ttl-note.mdx:
```diff
@@ -0,0 +1,5 @@
+<Admonition type="note" title="TTL">
+  We recommend using shorter TTL but keep mind that you'll need to update the
+  database server certificate before it expires to not lose the ability to
```
Great, thanks!

inertial-frame reviewed (changes requested):
Overall this is great and already addresses some feedback from the Validation Pass! 

Vastly improved both in terms of layout and brevity. (Akin to a preference for poems rather than anthologies.) Kudos!

Most suggestions here are editorial. Key questions involve icon use, backporting, and planning for future UI/UX components.

Will also validate but wanted to get a few comments in beforehand.

inertial-frame commented on docs/pages/database-access/guides/mongodb-self-hosted.mdx:
```diff
@@ -0,0 +1,348 @@
+
+<Tabs>
+  <TabItem label="Standalone">
+  When connecting to standalone MongoDB, sign certificate for the hostname over
```
May want to double-check this component. Looks like a tag is unclosed.

r0mant replied:
It is closed, renders fine also.

inertial-frame commented on docs/pages/database-access/guides/mysql-self-hosted.mdx:
```diff
@@ -7,54 +7,20 @@ description: How to configure Teleport Database Access with self-hosted MySQL
 
```
What do you think about adding:

```
(!docs/pages/includes/permission-warning.mdx!)

(!docs/pages/includes/backup-warning.mdx!)

## Create role and user

(!docs/pages/database-access/includes/db-user-roles.mdx!)
```

inertial-frame replied:
We recommend creating at least one Teleport role specific for database access. 

And, users that wish to connect to a database through Teleport will want to have the ability to [impersonate](https://goteleport.com/docs/access-controls/guides/impersonation/) that role.

Below, we'll create both of these Role Templates to allow a user to connect to any database using any database account in one handy command:

```bash
cat > dbroles.yaml << 'EOF'
kind: role
version: v3
metadata:
  name: db
spec:
  allow:
    db_labels:
      '*': '*'
    db_names:
    - '*'
    db_users:
    - '*'
---
kind: role
version: v3
metadata:
  name: impdbrole
spec:
  allow:
    impersonate:
      users: ["Db"]
      roles: ["Db"]
EOF
```

```bash
tctl create -f dbroles.yaml
```

Create a user `db-user` with the defined database access like so:

```bash
tctl users add db-user --roles=access,impdbrole --logins=root,ubuntu,ec2-user
```

inertial-frame replied:
The include ^ - I found that to be a bit confusing.

r0mant replied:
I think this will break the flow of these guides. In general the way I currently designed them is:

1. First users configure their database instance (self-hosted or cloud).
2. Then they configure Teleport.

As such, creating Teleport users/roles in the beginning doesn't really make sense.

In addition there are a couple of inaccuracies in the version you proposed:

* Users don't have to impersonate `Db` to connect. It's only needed to generate secrets for self-hosted dbs in some cases (e.g. Cloud) - which is why it's at the beginning.
* These particular instructions are actually a bit obsolete - for new guides we don't ask them to create roles and use builtin `access` role we have. I need to update older guides as well but it's out of scope of this PR.

In general I'd like to move Teleport user creation to a partial like you suggested, though the difficulty is details differ a little bit between guides for different databases so I'm not sure yet best way to do it. I'll come back to it later, it's out of scope of this PR.

inertial-frame replied:
The layout is currently a brief description about Teleport with several commands describe (although these are not explicit instructions or steps), then a partial walkthrough about configuring the database (with most steps omitted), then a bit about Teleport, then some more about the database(s), then some more about Teleport locally.

That's _verbatim_ what's on the page. Apart from that, in practice, the Guide interleaves multiple separate configuration steps and is difficult to complete.  Don't take my subjective opinion as the primary criteria for evaluation here:

_Bounce rates are 20% higher in this Section and it has the 4th highest exit rate of all content._

r0mant replied:
@inertial-frame What do you propose?

inertial-frame replied:
Don't mean for this to turn into a blocker (fan of the idea that we should merge changes if the PR's represent something non-breaking and strictly better even if there's something incomplete or could be improved - for example adding a section but not all of the planned articles, or updating some document even if everything's not 100% decided). 

I also want to avoid technical and editorial debt. We also have at least 2 weeks (and then some) before the 7.0 release (great job getting these changes in so early)!

---

For starters, I walked through each of these and documented the steps I needed to take to advance from beginning to end: https://docs.google.com/document/d/1C8CT-JSFpRcxPxfP1zX4a-LwTb5NKlWyG0mk4f6q5MI/edit# - I found these to be time-consuming and challenging - not necessarily because the articles were difficult but because configuring and installing a database is complex in its own right (plus all of the other items on top)!

The Teleport steps are actually the breeziest (again kudos on the great work). 

In walking through those steps, the one thing I found to be something we could improve on is the Teleport flow. We introduce a concept at the beginning but return to it later on. Also, no explicit example is given (a few small steps are).

That seemed to vastly simplify this process for me - it was not clear what roles were required, that the created user should be given the impersonation role, etc. The other items involved figuring out the best way to configure the DB's.

These opinions are not lines in the sand but I do think that step needs to be improved. Arguably, writing and reading a setting up DB's in a self-hosted capacity article (whether locally, on EC2, in a VM) should involve less domain knowledge (since you don't need to describe a lot of the IAM and configuration items - these are definite musts in production but can typically be skipped in an article whereas cloud configuration settings typically must be conveyed in an article - the latter is probably easier to use and get set up but the article format involves describing many additional pieces).

r0mant replied:
Yeah I agree that instructions on how to install a self-hosted database should probably not be a part of these guides - each system is different and we wouldn't be able to cover them all. Also, installing a database server is not actually the part folks struggle with the most (usually, IAM stuff is the most tricky).

One issue about the current self-hosted flow that I've just realized is that we ask the users to use `tctl auth sign` to produce certificates - which requires a Teleport cluster, but the instructions of installing Teleport actually follow much later... We could tweak it a bit like this:

1. Install and configure Teleport Auth/Proxy - OR - Instructions for Teleport Cloud users.
2. Configure Postgres/MySQL database, which involves:
  2.1 Using `tctl auth sign` to produce secrets.
  2.2 Use those secrets to configure mTLS on the database.
  2.3 Create database user, etc.
3. Then proceed to configuring Teleport Database service.
4. Finally, connect.

So basically installing/configuring Teleport cluster moves to first step and the rest follows - one thing kinds of leads to another. What do you think? I could try and apply this change to the new Mongo guide here (since it's basically self-hosted) and do other guides later as well.

r0mant replied:
@inertial-frame I have refactored MongoDB guide. I think it turned out quite nicer now, I rearranged the flow as described above, simplified some things and moved some things into partials. Here's a brief summary:

* Explicitly mentioned steps and prerequisites at the beginning like other guides do.
* Rearranged the flow to start with Teleport - it flows much better now (IMO).
* Made "create user" section more generic and moved into a partial - so we can reuse in other guides too.
* Removed "start with configuration file" section entirely - I think it was only complicating things.
* Added more clarify around tctl auth usage for self-hosted Teleport vs Cloud.

Here's the commit with the changes: https://github.com/gravitational/teleport/pull/7350/commits/d3b8ebc903553f186c8c9c7e2dfc3833845aa4cd. Let me know what you think.

inertial-frame commented on docs/pages/database-access/guides/mysql-self-hosted.mdx:
````diff
@@ -87,53 +53,11 @@ GRANT ALL ON `%`.* TO 'alice'@'%';
-```
+(!docs/pages/includes/database-access/start-auth-proxy.mdx!)
 
 ### Create Role and User
````
Per the comment above, we could probably remove this section if we moved the include to the beginning. What do you think? 

We'd repeat this for both of the self-hosted articles.

r0mant replied:
Replied above, these can't go at the beginning because these are a part of Teleport setup.

inertial-frame commented:
One last request 🙏🏼  - do you mind also updating the GUI article in this PR?

I found the following layout to be improved. What do you think?

---

---
title: Database Access GUI Clients
description: How to configure graphical database clients for Teleport Database Access.
---

# Graphical Database Clients

This article describes how to configure popular graphical database clients to work with Teleport Database Access.

## Obtain local certificate/key-pair

Issue the following command after you login through `tsh`:

```bash
tsh db config
tsh db config --db=example
```

It will display the path to your locally cached PEM files:

```
# Output
Name:      aurora
Host:      root.gravitational.io
Port:      3036
User:      r0mant
Database:
CA:        /Users/r0mant/.tsh/keys/root.gravitational.io/certs.pem
Cert:      /Users/r0mant/.tsh/keys/root.gravitational.io/r0mant-db/root/aurora-x509.pem
Key:       /Users/r0mant/.tsh/keys/root.gravitational.io/r0mant
```

The displayed `CA`, `Cert`, and `Key` files are used to connect through pgAdmin 4 and MySQL Workbench.

## pgAdmin 4

[pgAdmin 4](https://www.pgadmin.org/) is a popular graphical client for PostgreSQL servers.

To configure a new connection, right-click on "Servers" in the main browser view and create a new server:

![pgAdmin Add Server](../../../img/database-access/pgadmin-add-server@2x.png)

In the "General" tab of the new server dialog, enter the server connection name:

![pgAdmin General](../../../img/database-access/pgadmin-general@2x.png)

In the "Connection" tab, enter the default database name (the *maintenance database*) and the connection service name (the same one that you specify when connecting through `psql`). Leave all the other fields blank:

![pgAdmin Connection](../../../img/database-access/pgadmin-connection@2x.png)

In the "SSL" tab, set "SSL Mode" to `Verify-Full`:

![pgAdmin SSL](../../../img/database-access/pgadmin-ssl@2x.png)

Click "Save", and pgAdmin should immediately connect.

## MySQL Workbench

[MySQL Workbench](https://www.mysql.com/products/workbench/) is a visual tool that provides comprehensive MySQL administration and SQL development tools.

In MySQL Workbench "Setup New Connection" dialog, fill out "Connection Name", "Hostname", "Port", and "Username":

![MySQL Workbench Parameters](../../../img/database-access/workbench-parameters@2x.png)

In the "SSL" tab, set "Use SSL" to `Require and Verify Identity` and enter paths to  your CA, certificate, and private key files from `tsh db config` command:

![MySQL Workbench SSL](../../../img/database-access/workbench-ssl@2x.png)

Optionally, click "Test Connection" to verify connectivity:

![MySQL Workbench Test](../../../img/database-access/workbench-test@2x.png)

Save the connection, and connect to the database.

r0mant commented on docs/config.json:
```diff
@@ -114,11 +114,14 @@
           "entries": [
             { "title": "AWS RDS/Aurora PostgreSQL", "slug": "/database-access/guides/postgres-aws/" },
             { "title": "AWS RDS/Aurora MySQL", "slug": "/database-access/guides/mysql-aws/" },
-            { "title": "AWS Redshift PostgreSQL", "slug": "/database-access/guides/postgres-redshift/" },
```
Nope, these are for 7.0 only.

r0mant commented on docs/pages/config-reference.mdx:
```diff
@@ -518,7 +518,7 @@ db_service:
     # Free-form description of the database proxy instance.
     description: "Production database"
-    # Database protocol. Can be "postgres" or "mysql".
+    # Database protocol. Can be "postgres", "mysql" or "mongodb".
```
TIL :)

r0mant commented on docs/pages/database-access/guides.mdx:
```diff
@@ -4,10 +4,49 @@ description: Guides for configuring Teleport Database Access with self-hosted or
-- [GUI clients](./guides/gui-clients.mdx)
+# Database Access Guides
+
+## Cloud-Hosted Guides
```
I went ahead and made this casing change in all of database access pages.

r0mant commented on docs/pages/database-access/guides.mdx:
```diff
@@ -4,10 +4,49 @@ description: Guides for configuring Teleport Database Access with self-hosted or
+## General Guides
+
+<TileSet>
+  <Tile icon="window" title="GUI clients" href="./guides/gui-clients.mdx">
```
Yeah, I picked them on purpose. For cloud-hosted guides the icon is "cloud", for self-hosted - "database", for GUI clients the "window" icon looks like a graphical application so seemed fitting.

r0mant commented on docs/pages/database-access/guides/ha.mdx:
```diff
@@ -0,0 +1,92 @@
+---
+title: Database Access High Availability
+description: How to configure Teleport Database Access in HA configuration
```
Gotcha, updated to "High Availability (HA)" everywhere and also added periods at the end of descriptions on all database access pages.

r0mant commented:
@inertial-frame Addressed your comments, here's the commit if you'd like to take a look at the changeset only: https://github.com/gravitational/teleport/pull/7350/commits/346c4ead53792ba56a97d3b9325c2af8b9deae8b. I've vendored your version of GUI Guides page as well.

inertial-frame reviewed (changes requested):
Please see my previous comments. Thanks!

r0mant commented:
@inertial-frame Which ones? I think I either addressed or replied to them, did I miss anything?

benarent commented on docs/pages/database-access/guides/gui-clients.mdx:
```diff
@@ -1,78 +1,84 @@
 ---
 title: Database Access GUI Clients
-description: How to configure graphical database clients
+description: How to configure graphical database clients for Teleport Database Access.
```
Does MongoDB setup work this https://www.mongodb.com/try/download/compass?

r0mant replied:
Kind of. I was able to connect but found out that it doesn't work well with our RBAC - i.e. if your Teleport role prohibits access to specific database, for some reason Compass refuses to load anything. It works otherwise - I'll see if we can maybe return a more friendly access denied error to make it work.

I also tried some other client called Robo 3T (https://robomongo.org/) - this one works fine. Not sure if it's popular or not though.

benarent replied:
That's good to know. No need to add anything now, but it'll be interesting to see what the community requests.

benarent commented on docs/pages/database-access/guides/mysql-cloudsql.mdx:
```diff
@@ -0,0 +1,254 @@
+---
+
+# GCP Cloud SQL MySQL
+
```
It would be nice to have a high level diagram to show people what the setup is needed.

benarent reviewed (approved):
The PR looks good.. setup still feels little complicated but I'll record videos and provide more UX / Setup feedback after.  A few initial setup diagrams might help.

inertial-frame reviewed (approved):
@r0mant - thanks for making the recommended changes - actually you made them even better than I suggested! We can also tweak these later on as needed.

Some other refactoring PR's along with some new Next components may help us improve a lot of our documentation, snippets, outputs, and so on. You won't need to worry about making those changes but I'll likely defer to you for your reviews, suggestions, etc.

Thanks again!

klizhentas reviewed (approved):
a couple of typos, otherwise LGTM

r0mant committed:
Fix linter
